internal/cmd/gocdk/launch.go

```diff
@@ -15,11 +15,170 @@
 package main
 
 import (
+	"bytes"
 	"context"
+	"flag"
+	"fmt"
+	"log"
+	"net/url"
+	"os/exec"
+	"path/filepath"
 
 	"golang.org/x/xerrors"
 )
 
 func launch(ctx context.Context, pctx *processContext, args []string) error {
-	return xerrors.New("not implemented")
+	f := newFlagSet(pctx, "launch")
+	imageName := f.String("image", "", "Docker image name to launch (defaults to project latest)")
```

Comment:
Definitely, I think the biome/launcher concepts will be useful with non-gocdk projects. I'm cool with having users specify both the name and the tag, just not just the name.

Reply:
Now I understand. I'm worried about the breaking of convention: when specifying Docker image name + tag, the `name[:tag]` form is prevailing. I've never seen a system that accepts `[name:]tag` or `[name]:tag` as input. Even `docker build -t` (the "tag" flag) takes in the `name[:tag]` form. We could separate this out as two different flags, but that can have surprising effects for the user.

It may be nice for `gocdk deploy` to have such a tag flag, since that would affect both `build` and `launch`, but I don't know if that's in scope for this PR. Can we open an issue for this and proceed from there?

Reply:
Did a little more digging on this, since I want to consider the effect on `gocdk build`. The definitive grammar for a Docker image reference is here: https://github.com/docker/distribution/blob/master/reference/reference.go

So in examining that, `:tag` would never be an acceptable image reference, so we could permit that short form to take the name from the project Dockerfile. Implemented that, WDYT?